### ISSUE TITLE
CT unable to find node from unique Trail id

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/circuit-construction-kit-dc#190]
```
circuit-construction-kit-dc : phet-io-mirror-inputs-fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/phet-io-wrappers/mirror-inputs/?sim=circuit-construction-kit-dc&phetioDebug&fuzz&wrapperContinuousTest=%7B%22test%22%3A%5B%22circuit-construction-kit-dc%22%2C%22phet-io-mirror-inputs-fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1613954877939%22%2C%22timestamp%22%3A1613963912352%7D
Uncaught Error: Assertion failed: unable to find node from unique Trail id
Error: Assertion failed: unable to find node from unique Trail id
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/assert/js/assert.js:25:13)
at Function.fromUniqueId (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/scenery/js/util/Trail.js:1292:19)
at PDOMPointer.invalidateTrail (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/scenery/js/input/PDOMPointer.js:115:26)
at Input.focusoutAction.Action.phetioPlayback (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/scenery/js/input/Input.js:590:26)
at Action.execute (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/axon/js/Action.js:225:18)
at Action.implementation (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/axon/js/Action.js:252:26)
at PhetioCommandProcessor.processCommand (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/phet-io/js/phetioCommandProcessor.js:291:51)
at https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/phet-io/js/phetioCommandProcessor.js:175:36
at Array.map (<anonymous>)
at PhetioCommandProcessor.processCommands (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/phet-io/js/phetioCommandProcessor.js:173:30)
id: Bayes Chrome
Snapshot from 2/21/2021, 5:47:57 PM

----------------------------------

circuit-construction-kit-dc : phet-io-state-fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/phet-io-wrappers/state/?sim=circuit-construction-kit-dc&phetioDebug&fuzz&wrapperContinuousTest=%7B%22test%22%3A%5B%22circuit-construction-kit-dc%22%2C%22phet-io-state-fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1613954877939%22%2C%22timestamp%22%3A1613959350826%7D
Uncaught Error: Assertion failed: trying to set focus on a node that is not focusable
Error: Assertion failed: trying to set focus on a node that is not focusable
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/assert/js/assert.js:25:13)
at Node.focus (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/scenery/js/accessibility/pdom/ParallelDOM.js:537:21)
at Object.afterOp (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/scenery/js/accessibility/pdom/PDOMTree.js:421:32)
at Object.pdomContentChange (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/scenery/js/accessibility/pdom/PDOMTree.js:227:14)
at VertexNode.onPDOMContentChange (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/scenery/js/accessibility/pdom/ParallelDOM.js:2707:18)
at VertexNode.setTagName (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/scenery/js/accessibility/pdom/ParallelDOM.js:884:16)
at VertexNode.set tagName [as tagName] (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/scenery/js/accessibility/pdom/ParallelDOM.js:887:37)
at https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/scenery/js/nodes/Node.js:6453:23
at arrayEach (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/sherpa/lib/lodash-4.17.4.js:537:11)
at Function.forEach (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1613954877939/sherpa/lib/lodash-4.17.4.js:9359:14)
id: Bayes Chrome
Snapshot from 2/21/2021, 5:47:57 PM
```

**Comment by samreid** [comment]
This sounds related to a11y/phet-io and focus.  @jessegreenberg and/or @zepumph can you please take a look?

**Comment by jessegreenberg** [comment]
It looks like this is only happening in the mirror inputs and state wrappers on CT. I wasn't able to get this to happen with PhET brand fuzzing locally. I also ran fuzzing in the state wrapper for about 5 minutes and didn't see the bug happen. I am wondering if this is related to https://github.com/phetsims/phet-io/issues/1720, that involved pdom/phet-io overlap.

**Comment by jessegreenberg** [comment]
I let the sim fuzz in mirror inputs and state wrapper for about 10 minutes each and then tried to create and delete multiple VertexNodes in each of these wrappers. But I wasn't able to see the error. I am not sure where to go next on PhET-iO side so removing my assignment, but feel free to pass it back.

**Comment by KatieWoe** [comment]
```
circuit-construction-kit-dc : phet-io-state-fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1614723829308/phet-io-wrappers/state/?sim=circuit-construction-kit-dc&phetioDebug&fuzz&wrapperContinuousTest=%7B%22test%22%3A%5B%22circuit-construction-kit-dc%22%2C%22phet-io-state-fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1614723829308%22%2C%22timestamp%22%3A1614726505791%7D
Uncaught Error: Assertion failed: trying to set focus on a node that is not focusable
Error: Assertion failed: trying to set focus on a node that is not focusable
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1614723829308/assert/js/assert.js:25:13)
at Node.focus (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1614723829308/scenery/js/accessibility/pdom/ParallelDOM.js:537:21)
at Object.afterOp (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1614723829308/scenery/js/accessibility/pdom/PDOMTree.js:421:32)
at Object.addChild (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1614723829308/scenery/js/accessibility/pdom/PDOMTree.js:41:14)
at CircuitElementEditContainerNode.onPDOMAddChild (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1614723829308/scenery/js/accessibility/pdom/ParallelDOM.js:2750:18)
at CircuitElementEditContainerNode.insertChild (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1614723829308/scenery/js/nodes/Node.js:633:12)
at CircuitElementEditContainerNode.addChild (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1614723829308/scenery/js/nodes/Node.js:661:10)
at https://bayes.colorado.edu/continuous-testing/ct-snapshots/1614723829308/circuit-construction-kit-common/js/view/CircuitElementEditContainerNode.js:343:14
at TinyEmitter.emit (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1614723829308/axon/js/TinyEmitter.js:86:18)
at Property._notifyListeners (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1614723829308/axon/js/Property.js:294:25)
id: Bayes Chrome
Snapshot from 3/2/2021, 3:23:49 PM
```

**Comment by zepumph** [comment]
I feel a bit blocked by this because CT and local testing seems to always hit "Error: Assertion failed: phetioID already registered: circuitConstructionKitDc.introScreen.view.circuitElementToolbox.carousel.tools.wireToolNode.lifelikeIcon" before I can get to an a11y error. @samreid what do you think?

**Comment by samreid** [comment]
Thanks, I think I corrected the upstream problem, let me know if that helps.

**Comment by zepumph** [comment]
Initial findings:

The delete key doesn't work in Mirror Inputs wrapper. It does work in the state wrapper and in the main sim. It also works well when recording to a local file and playing back.

**Comment by zepumph** [comment]
TLDR: I'm leaning towards thinking of this bug as a dynamic elements problem. But also its cause is most likely because of our lack of support for document focus manipulation in mirror inputs.

With this patch and fuzzing mirror inputs you can get a debugger right where the assertion occurs. Sometimes it took ~2 minutes of fuzzing to reproduce. (?fuzz)

```diff
Index: js/util/Trail.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/util/Trail.js b/js/util/Trail.js
--- a/js/util/Trail.js	(revision 90d2b7380a66383c29fa7cf6f7baed3b4a3130f5)
+++ b/js/util/Trail.js	(date 1615311375170)
@@ -1283,7 +1283,11 @@
           break;
         }
 
-        assert && assert( j !== children.length - 1, 'unable to find node from unique Trail id' );
+        const b = j !== children.length - 1;
+        if( !b){
+          debugger;
+        }
+        assert && assert( b, 'unable to find node from unique Trail id' );
       }
     }
 
```

Once there, I investigated the current Trail that was trying to be reconstituted by the PDOM id:

"230-231-584-593-604-4680"


Here is my console (sorry for just the picture):
![image](https://user-images.githubusercontent.com/6856943/110514817-277e7680-80b4-11eb-8cfe-8e16fe843143.png)

We are paused in the downstream sim, trying to execute a focusIn event on a vertex. But it (Node#4680) seems like it doesn't exist.


@samreid, in the mirror inputs wrapper it is very hard to tell when things get out of sync. That said, we know that focus isn't supported with the tool. It is most likely that the PDOM events which change focus (and therefore change `document.activeElement`) are altering the playback and making things get out of sync. 

That is my best guess, but you are more in tune with the workings of this sim. What do you think?

**Comment by samreid** [comment]
I fuzzed mirror inputs with the patch and got the same assertion.  I didn't have any insight from the context in the debugger.  I'm not sure how to proceed--should we disable focus, keyboard or accessibility for the mirror inputs wrapper?

**Comment by zepumph** [comment]
It seems like this is a potential problem related with https://github.com/phetsims/phet-io/issues/1748.

**Comment by zepumph** [comment]
@samreid, are you still having trouble with this?

**Comment by matthew-blackman** [comment]
This hasn't shown up on CT, and we no longer have a mirror inputs wrapper. @samreid and I feel that issue can be closed.
